vortex-array/src/arrow/record_batch.rs

```diff
@@ -30,7 +33,9 @@ impl StructArray {
 
     pub fn into_record_batch_with_schema(self, schema: &Schema) -> VortexResult<RecordBatch> {
         let data_type = DataType::Struct(schema.fields.clone());
-        let array_ref = to_arrow(self.as_ref(), &data_type)?;
+        let dtype = DType::from_arrow((&data_type, Nullability::NonNullable));
+        let casted = cast(self.as_ref(), &dtype)?;
```

Comment:
this cast seems fishy

Reply:
I'm all for doing it well, I'll look into it. I'm in no rush to merge any of this.